assets/dev/js/editor/regions/panel/pages/elements/views/elements.js

```diff
@@ -22,12 +22,17 @@ PanelElementsElementsView = Marionette.CollectionView.extend( {
 			return false;
 		}
 
-		// Remove widgets from search results by 'widgets categories'.
-		const showInSearchResult = Object.keys( elementor.documents.getCurrent().config.panel.elements_categories )
-			.some( ( categoryName ) => childModel.get( 'categories' ).includes( categoryName ) );
+		const isGlobal = childModel.get( 'custom' )?.templateID;
 
-		if ( ! showInSearchResult ) {
-			return false;
+		// Filter Results only on widgets which are not global widget.
+		if ( ! isGlobal ) {
```

Comment:
Please add a flag `show_on_search` for categories & widgets that shouldn't appear in the search results